### PR TITLE
Gps odom set up

### PR DESCRIPTION
**Pull request by Quinnan-Gill**
Hi, so I am just curious if I set this up correctly. I have not tested anything.

### CODE REVIEW

File: src/magellan_core/launch/gps.launch [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<launch>
+    <node pkg="nmea_navsat_driver" type="nmea_serial_driver" name="navsat" respawn="true>
+        <param name="port" value="/dev/ttyUSB0"/>
+        <param name+"baud" value="19200/>
```
Comment (asaba96) [inline src/magellan_core/launch/gps.launch]: I believe the plus here should be an equal sign

File: src/gps_odom/src/Conversions.h [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+
+    double T = tlat * tlat;
+    double C = UTM_EP2 * clat * clat;
+    double A = (rlon - rlon0) * clat;
+    double M = WGS84_A * (m0 * rlat + m1 * sin(2 * rlat)
+                          + m2 * sin(4 * rlat) + m3 * sin(6 * rlat));
+    double V = WGS84_A / sqrt(1 - UTM_E2 * slat * slat);
+
+    // compute the easting-northing coordinates
+    *x = UTM_FE + UTM_K0 * V * (A + (1 - T + C) * pow(A,3) / 6
```
Comment (asaba96) [inline src/gps_odom/src/Conversions.h]: I know the linter formatted this file and passed it, but is there anything we can do to clean up this math while keeping the linter happy/ @4ndr3w thoughts?

File: src/gps_odom/launch/GPS_Odom.launch [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<launch>
```
Comment (asaba96) [inline src/gps_odom/launch/GPS_Odom.launch]: @4ndr3w whats our naming convention on launch file names?

**Comment by asaba96** [comment]
Also, we should get this deployed to the bot and get it tested

File: src/magellan_core/launch/gps.launch (hunk 1 above)
Comment (Quinnan-Gill) [reply src/magellan_core/launch/gps.launch]: Fixed type

File: src/gps_odom/src/Conversions.h (hunk 2 above)
Comment (Quinnan-Gill) [reply src/gps_odom/src/Conversions.h]: Put it through a formatter

File: src/magellan_core/launch/passive.launch [hunk 4]
```diff
@@ -7,8 +7,10 @@
     <node pkg="magellan_core" type="viz_turning_radius.py" name="viz_turning_radius" />
 
     <include file="$(find magellan_core)/launch/localization.launch" />
-    <include file="$(find magellan_core)/launch/lidar.launch" />
+    <!--<include file="$(find magellan_core)/launch/lidar.launch" />-->
```
Comment (4ndr3w) [inline src/magellan_core/launch/passive.launch]: Can we check this in with lidar uncommented?

File: src/magellan_core/launch/robot_localization.launch [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+                                     true, true, true]</rosparam>
+
+      <param name="odom0_differential" value="true"/>
+      <param name="imu0_differential" value="true"/>
+
+      <param name="imu0_remove_gravitational_acceleration" value="false"/>
+
+      <rosparam param="process_noise_covariance">[LARGE ARRAY with small values]</rosparam>
+
+      <rosparam param="initial_estimate_covariance">[LARGE ARRAY with 1e-9]</rosparam>
```
Comment (4ndr3w) [inline src/magellan_core/launch/robot_localization.launch]: This seems like example text - is this file loaded anywhere?

File: src/magellan_core/params/odom_ekf.yaml [hunk 6]
```diff
@@ -121,6 +119,13 @@ stamped_control: false
 control_timeout: 0.2
 # Which velocities are being controlled. Order is vx, vy, vz, vroll, vpitch, vyaw.
 control_config: [true, false, false, false, false, true]
+# Whether the input (assumed to be cmd_vel) is a geometry_msgs/Twist or geometry_msgs/TwistStamped message. Defaults to
+# false.
+stamped_control: false
+# The last issued control command will be used in prediction for this period. Defaults to 0.2.
+control_timeout: 0.2
+# Which velocities are being controlled. Order is vx, vy, vz, vroll, vpitch, vyaw.
+control_config: [true, false, false, false, false, true]
```
Comment (4ndr3w) [inline src/magellan_core/params/odom_ekf.yaml]: Why are we fusing control inputs? I don't think we provide that as a topic right now

File: src/my_package/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,199 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(my_package)
```
Comment (4ndr3w) [inline src/my_package/CMakeLists.txt]: Looks like an example package slipped into the commit?

File: src/magellan_core/launch/passive.launch (hunk 4 above)
Comment (Quinnan-Gill) [reply src/magellan_core/launch/passive.launch]: Uncommented

File: src/magellan_core/launch/robot_localization.launch (hunk 5 above)
Comment (Quinnan-Gill) [reply src/magellan_core/launch/robot_localization.launch]: Sorry this is file is not needed anymore. I made when I was first trying to figure out robot_localization

File: src/my_package/CMakeLists.txt (hunk 7 above)
Comment (Quinnan-Gill) [reply src/my_package/CMakeLists.txt]: Deleted

File: src/magellan_core/params/odom_ekf.yaml (hunk 6 above)
Comment (Quinnan-Gill) [reply src/magellan_core/params/odom_ekf.yaml]: Set all of them to false

**Comment by 4ndr3w** [comment]
Closing, pulled GPS in #65

**Pull request closed without merging**